# This file is part of HiGFXback

# requires
REQUIRES="libdrm-build make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix

# build
make

# install
install -d $DESTDIR/bin
install drm-framebuffer $DESTDIR/bin

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/drm-framebuffer-build.pc << EOF
Name: drm-framebuffer
Version: 20231108
Description: Utility to pipe data to a DRM dumb framebuffer
Requires: $REQUIRES

devel=\\
/bin/drm-framebuffer
EOF

exit
--- drm-framebuffer-20231108.orig/Configure
+++ drm-framebuffer-20231108/Configure
@@ -0,0 +1,29 @@
+#!/bin/sh
+
+prefix=/usr/local
+
+while : ; do
+  case "$1" in
+    "")
+      break
+    ;;
+    --prefix)
+      prefix=$2
+      shift
+    ;;
+    *)
+      echo Unknown option \"$1\".
+      exit 1
+    ;;
+  esac
+  shift
+done
+
+sed -i "/^prefix/s|=.*|= $prefix|" Makefile
+
+test $PKG_CONFIG || PKG_CONFIG=pkg-config
+if test $PKG_CONFIG_PATH; then
+  sed -i "/^PKG_CONFIG/s|=.*|= PKG_CONFIG_PATH=$PKG_CONFIG_PATH $PKG_CONFIG|" Makefile
+else
+  sed -i "/^PKG_CONFIG/s|=.*|= $PKG_CONFIG|" Makefile
+fi
--- drm-framebuffer-20231108.orig/Makefile
+++ drm-framebuffer-20231108/Makefile
@@ -1,12 +1,14 @@
+PKG_CONFIG ?= pkg-config
+
 CC      ?= gcc
-CFLAGS  ?= -O0 -ggdb -pedantic -Wall -I /usr/include/libdrm
-LDFLAGS ?= -ldrm
+CFLAGS  ?= -O0 -ggdb -pedantic -Wall $(shell $(PKG_CONFIG) --cflags libdrm)
+LDFLAGS ?= $(shell $(PKG_CONFIG) --libs libdrm)
 
 OBJ = main.o framebuffer.o
 PROGNAME = drm-framebuffer
 
-exec_prefix ?= /usr
-bindir ?= $(exec_prefix)/bin
+prefix ?= /usr/local
+bindir ?= $(prefix)/bin
 
 all: $(OBJ)
 	$(CC) $(CFLAGS) -o $(PROGNAME) $(OBJ) $(LDFLAGS)
